_posts/2021-01-31-choosing-randomization-unit.md

```diff
@@ -0,0 +1,353 @@
+
+A key decision you must make with any A/B test is how you create these two groups. I'll use an e-commerce website as an example throughout the remainder of this post, but the concepts will readily apply to any type of website or user facing application. 
+
+Let's say you want to test out two different versions of your e-commerce website, and see which version is better for your business. We know we need to randomly divide our population into two equally sized groups, and give each of those groups a different experience. But how should it be split up?
+
+
+<p align="center">
+    <img width="70%" src="{{ site.baseurl }}{% link images/choosing-randomization-unit/population_split.png %}">
+</p>
+
+In the realm of web applications, three common ways of doing this are:
+
+- **Page views**: for each page that is rendered (shown to a person), randomly choose to use version A or version B for that particular page.
+- **Sessions**: sessions are just a grouping of page views made by the same person, typically bound by a specific time period. In session level randomization, we'll randomly choose to use version A or version B of the site for all page views in a given session.
+- **Users:** users are people, or a close approximation of them. If a user must log into your application, you can easily identify them with something like an account id. For most websites, you don't need to log in, so users are instead identified using [cookies](https://en.wikipedia.org/wiki/HTTP_cookie). Because a single person could clear their cookies, or use a different device or browser, and therefore get assigned a new cookie, these will only serve as an approximation. In user level randomization, we can randomly choose whether to show version A or version B to a particular "user" across all their sessions and page views.
+
+These are all examples of different **randomization units**. A randomization unit is the "who" or "what" that is randomly allocated to each group. Your choice of randomization unit will depend on multiple things. Technical feasibility will often be a limiting factor - do you have reliable ways of identifying sessions & users? The type of change you are making will also influence your choice. Are you testing out a new page layout and trying to see if it causes users to remain on a page for a longer period of time? Page view level randomization could be a good candidate in this case. What if you are testing out an entirely different color scheme for your site and seeing how it impacts your account signup rate? A different colour appearing on each page view could be a pretty jarring experience for users, so session or user level randomization would likely be a better choice in order to provide a consistent experience.
```

Comment:
```suggestion
These are all examples of different **randomization units**. A randomization unit is the "who" or "what" that is randomly allocated to each group. Your choice of randomization unit will depend on multiple things. Technical feasibility will often be a limiting factor - do you have reliable ways of identifying sessions & users? The type of change you are making will also influence your choice. Are you testing out a new page layout and trying to see if it causes users to remain on a page for a longer period of time? Page view level randomization could be a good candidate in this case. What if you are testing out an entirely different color scheme for your site and seeing how it impacts your account signup rate? A different colour appearing on each page view could be a pretty jarring experience for users, so session or user level randomization would likely be a better choice in order to provide a consistent experience. 

Your decision of randomization unit may also be influenced by the size of the effect you are trying to calculate and the available sample size. Choosing a more narrow grain like page views can give you more statistical power since you'll typically get a significantly larger sample size when randomizing at this level.
```